### PR TITLE
storage: Add store id to the pebble logs in cockroach

### LINKED ISSUE #64798: storage: pebbleLogger context should include store ID
The context captured when a `pebbleLogger` is created currently only contains the node ID. If a node has multiple stores, it can be difficult or impossible to disentangle which log messages are occurring for which store. It should be straightforward to make sure the context also includes a tag for the store ID. This would probably be done in `server.Config.CreateEngines`.

### PR DESCRIPTION
**Pull request by bananabrick**
This change adds the store id to the pebble logger. Since
the store id isn't available when the pebble logger is
created, a struct with an int32 is used as the store id
tag in the log, instead of an int32. The store id in the
struct is updated atomically later when the store id is
available.

Resolves: #64798, #65742

Release note: None

### CODE REVIEW

**Comment by bananabrick** [comment]
___
*[pkg/base/node_id.go, line 131 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpBBHcBNdip0ez8pji:-McpBBHcBNdip0ez8pjj:boci9x7) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac3c91c5857f4f5ad5f32b117962efcb6324c4b7/pkg/base/node_id.go#L131)):*
> ```Go
> 	if val <= 0 {
> 		if log.V(2) {
> 			log.Infof(ctx, "trying to set invalid StoreID for the Pebble log: %d", val)
> ```

Should these be log.Fatalf?

**Review by itsbilal — COMMENTED** [review]
Great work! This looks really solid, glad to see it happen :tada:

Most comments are minor suggestions, the only major one is about the addition to `PebbleConfig` and about getting around that.

> Reviewed 2 of 5 files at r1, 4 of 4 files at r2, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-:-McpR6adB7WvqLtp7T2M:b-glvlkb)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bananabrick)

___
*[pkg/base/node_id.go, line 79 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNo4v3KHmmTjP4SKC:-McpNo4v3KHmmTjP4SKD:bbfq38r) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L79)):*
> ```Go
> 
> // StoreIDPebbleLog is added as a logtag in the pebbleLogger's context.
> // The storeID value is later set atomically.
> ```

Nit: Justify the comment so the next line starts where this sentence ends.
___
*[pkg/base/node_id.go, line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNu6H3GUALmBLUMiG:-McpNu6I0Y8v4wulcloi:b939pn3) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L80)):*
> ```Go
> // StoreIDPebbleLog is added as a logtag in the pebbleLogger's context.
> // The storeID value is later set atomically.
> // The storeID can only be set later because the storeID is determined only
> ```

Nit: s/later/after engine creation/
___
*[pkg/base/node_id.go, line 82 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNYQk8ojS6eVSu8yG:-McpNYQk8ojS6eVSu8yH:b-9vntar) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L82)):*
> ```Go
> // The storeID can only be set later because the storeID is determined only
> // after the pebbleLogger's context is created.
> type StoreIDPebbleLog struct {
> ```

Might be worthwhile to just call this `StoreIDContainer` and have the comments continue saying it's meant to be a logtag.
___
*[pkg/base/node_id.go, line 90 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNi9m-67-vJzoYAdy:-McpNi9m-67-vJzoYAdz:b-psn5rt) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L90)):*
> ```Go
> }
> 
> const tempStoreIDPebbleLog = -1
> ```

You can export this (by uppercasing the first letter) so this can be used in the method where you create the temp engine. Could also be called `TempStoreContainerID`
___
*[pkg/base/node_id.go, line 137 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNNt3682yvDw19k7o:-McpNNt3682yvDw19k7p:b683sd2) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L137)):*
> ```Go
> 	if oldVal == 0 {
> 		if log.V(2) {
> 			log.Infof(ctx, "StoreID for the Pebble log set to %d", val)
> ```

This might be "too verbose" of a log entry as it'll happen every time a store starts up. You can leave the other ones in that represent errors, but remove this one as it's the happy path.
___
*[pkg/base/node_id.go, line 131 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpBBHcBNdip0ez8pji:-McpN8Bn9fm0fA-HtqpF:bn92nw9) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac3c91c5857f4f5ad5f32b117962efcb6324c4b7/pkg/base/node_id.go#L131)):*
<details><summary><i>Previously, bananabrick (Arjun Nair) wrote…</i></summary><blockquote>

Should these be log.Fatalf?
</blockquote></details>

Nope, info is fine. Its not in a critical path so fataling here might be an undesirable side effect.
___
*[pkg/base/node_id.go, line 95 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpOduJ-DQKUHa7ZKQm:-McpOduJ-DQKUHa7ZKQn:b-vvy0wg) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/base/node_id.go#L95)):*
> ```Go
> // either as a container for a temp store or a container for
> // an uninitialized storeID of 0, which will be initialized later.
> func InitStoreIDPebbleLog(tempStore bool) *StoreIDPebbleLog {
> ```

Could be `InitStoreIDContainer` and the bool removed so that the temp engine creator calls `Set(TempStoreID)`
___
*[pkg/kv/kvserver/store.go, line 1384 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpPnl67IM7ikRepc4x:-McpPnl67IM7ikRepc4y:bl5ypr1) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/kv/kvserver/store.go#L1384)):*
> ```Go
> 	log.Event(ctx, "read store identity")
> 
> 	// Set the store ID for Pebble logs if the engine is Pebble underneath the hood. This couldn't be set before because
> ```

I don't think knowledge of Pebble is necessary here. You could just have the comment say "communicate store ID to engine for logging if it needs it."

Also nit: wrap line at 80 chars
___
*[pkg/storage/pebble.go, line 426 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpBw710isOaJTsyk0t:-McpBw710isOaJTsyk0u:b-1ytjns) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/storage/pebble.go#L426)):*
> ```Go
> 	// If it's not nil, then NewPebble will add the StoreIDPebbleLog
> 	// to the logtags for its logger.
> 	PebbleStoreIDLogContainer *base.StoreIDPebbleLog
> ```

You don't need this here - the engine could create this. It's a struct with behaviour, and not a config variable, so it feels out of place.

You could either rely entirely on the setter method through the interface cast, like I suggested in the other comments. Or you could replace this with an `IsTempStore bool`, which `NewPebble` would read and do the right thing with, if it's set to true. That bool could also be a part of `base.StorageConfig` instead of `PebbleConfig`. I'll let you make the call.
___
*[pkg/storage/pebble.go, line 480 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpP6YPAEpN96zJVPis:-McpP6YPAEpN96zJVPit:b189hae) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/pebble.go#L480)):*
> ```Go
> 
> // LogStoreIDSetter is used to set the store id in the log.
> type LogStoreIDSetter interface {
> ```

Can just be `StoreIDSetter` and the method `SetStoreID`. 
___
*[pkg/storage/pebble.go, line 565 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpPJ086Cb93J_bSHbS:-McpPJ086Cb93J_bSHbT:b-y3n2dx) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/pebble.go#L565)):*
> ```Go
> 	logCtx := logtags.WithTags(context.Background(), logtags.FromContext(ctx))
> 	logCtx = logtags.AddTag(logCtx, "pebble", nil)
> 	if cfg.PebbleStoreIDLogContainer != nil {
> ```

You could create the container here, and add it onto the logger here as well.
___
*[pkg/storage/temp_engine.go, line 107 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpPYD5ALd5DYw1RylY:-McpPYD5ALd5DYw1RylZ:b2gxaeb) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/temp_engine.go#L107)):*
> ```Go
> 			StorageConfig:             storageConfig,
> 			Opts:                      opts,
> 			PebbleStoreIDLogContainer: base.InitStoreIDPebbleLog(true),
> ```

Replace this with a `p.SetStoreID(base.TempStoreID)` below.

**Comment by bananabrick** [comment]
___
*[pkg/base/node_id.go, line 82 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNYQk8ojS6eVSu8yG:-Mcpc2Pq-G141fA3WVvO:byyb4gr) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L82)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Might be worthwhile to just call this `StoreIDContainer` and have the comments continue saying it's meant to be a logtag.
</blockquote></details>

I called it this to limit its scope. This isn't like the `NodeIDContainer` which has a much larger scope.

**Review by bananabrick — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66728)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bananabrick and @itsbilal)

___
*[pkg/base/node_id.go, line 79 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNo4v3KHmmTjP4SKC:-McpcSta1mPRfgWVDPXv:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L79)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Nit: Justify the comment so the next line starts where this sentence ends.
</blockquote></details>

Done.
___
*[pkg/base/node_id.go, line 80 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNu6H3GUALmBLUMiG:-McpdSia8-KMdIP22ByT:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L80)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Nit: s/later/after engine creation/
</blockquote></details>

Done.
___
*[pkg/base/node_id.go, line 90 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNi9m-67-vJzoYAdy:-McpffiG028MR6TU9_el:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L90)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

You can export this (by uppercasing the first letter) so this can be used in the method where you create the temp engine. Could also be called `TempStoreContainerID`
</blockquote></details>

Done.
___
*[pkg/base/node_id.go, line 137 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNNt3682yvDw19k7o:-Mcpg9jh4CLrzfWCx5i-:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L137)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

This might be "too verbose" of a log entry as it'll happen every time a store starts up. You can leave the other ones in that represent errors, but remove this one as it's the happy path.
</blockquote></details>

Done.
___
*[pkg/base/node_id.go, line 95 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpOduJ-DQKUHa7ZKQm:-McpfoVi0D8Fhf2Dxhgn:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/base/node_id.go#L95)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Could be `InitStoreIDContainer` and the bool removed so that the temp engine creator calls `Set(TempStoreID)`
</blockquote></details>

Done.
___
*[pkg/storage/pebble.go, line 426 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpBw710isOaJTsyk0t:-McpiqJD380zgwzdgZgn:bk9cr20) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/storage/pebble.go#L426)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

You don't need this here - the engine could create this. It's a struct with behaviour, and not a config variable, so it feels out of place.

You could either rely entirely on the setter method through the interface cast, like I suggested in the other comments. Or you could replace this with an `IsTempStore bool`, which `NewPebble` would read and do the right thing with, if it's set to true. That bool could also be a part of `base.StorageConfig` instead of `PebbleConfig`. I'll let you make the call.
</blockquote></details>

Done. I let `NewPebble` create an uninitialized container and then I set it later for both temp/main stores.
___
*[pkg/storage/pebble.go, line 480 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpP6YPAEpN96zJVPis:-McpemD46hBd4cRTHxqM:b-pmrjav) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/pebble.go#L480)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Can just be `StoreIDSetter` and the method `SetStoreID`. 
</blockquote></details>

I wanted to limit any misunderstandings of what this was doing. I think having log in there is pretty useful as it makes it clear that we're setting the store ID only for the log. Let me know what you think!
___
*[pkg/storage/pebble.go, line 565 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpPJ086Cb93J_bSHbS:-McpjUuc9bQ3L9PPhxjm:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/pebble.go#L565)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

You could create the container here, and add it onto the logger here as well.
</blockquote></details>

Done.
___
*[pkg/storage/temp_engine.go, line 107 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpPYD5ALd5DYw1RylY:-McpjWR14MvqmzhT5FF8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/temp_engine.go#L107)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Replace this with a `p.SetStoreID(base.TempStoreID)` below.
</blockquote></details>

Done.

**Review by itsbilal — CHANGES_REQUESTED** [review]
Looks great! Just one last set of comments and then this should be good to go.

> Reviewed 6 of 6 files at r4.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-:-MctDObnAxxQ4vQN-iMw:b-a0lnak)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @bananabrick and @itsbilal)

___
*[pkg/base/node_id.go, line 82 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNYQk8ojS6eVSu8yG:-MctC27rDecXKQd4X5Wx:be3qsbm) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L82)):*
<details><summary><i>Previously, bananabrick (Arjun Nair) wrote…</i></summary><blockquote>

I called it this to limit its scope. This isn't like the `NodeIDContainer` which has a much larger scope.
</blockquote></details>

Right, but that's just how it's being used. Functionally it's very similar to `NodeIDContainer` so there's no reason to add extra details here.
___
*[pkg/storage/pebble.go, line 480 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpP6YPAEpN96zJVPis:-MctCo305r9h2ZhHmNt_:b7mlhxp) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/pebble.go#L480)):*
<details><summary><i>Previously, bananabrick (Arjun Nair) wrote…</i></summary><blockquote>

I wanted to limit any misunderstandings of what this was doing. I think having log in there is pretty useful as it makes it clear that we're setting the store ID only for the log. Let me know what you think!
</blockquote></details>

I see what you mean, but it's just requiring the caller to know extra info. All the caller needs to know is "the engine needs a store ID" and the engine's job is "I need to put the store ID in the log's tags". Helps to consolidate logic in one place instead of putting things in multiple places. So yes, I'd advise renaming the method to `SetStoreID`, the interface to `StoreIDSetter`, the init function to `InitStoreIDContainer` and the struct to `StoreIDContainer`. That way if we want to reuse this code for multiple things (eg. sentry tags), it's pretty intuitively clear what each piece is individually doing.
___
*[pkg/storage/temp_engine.go, line 110 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-MctCaenC6N1jHAGU42L:-MctCaenC6N1jHAGU42M:bb33p08) ([raw file](https://github.com/cockroachdb/cockroach/blob/30ddf5cb5708f53e911c70cbbae75877a42014a8/pkg/storage/temp_engine.go#L110)):*
> ```Go
> 	)
> 	// Set store ID for the pebble engine.
> 	LogStoreIDSetter(p).SetStoreIDForLog(ctx, base.TempStoreIDPebbleLog)
> ```

The usual convention is to type this as `p.(interfaceName).methodName(...)`.  Also this would have to be after the error check below, as if err != nil, p == nil and this would throw a panic.

**Review by bananabrick — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66728)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @itsbilal)

___
*[pkg/base/node_id.go, line 82 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpNYQk8ojS6eVSu8yG:-MctKzWm7rCe41YFspHI:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/1732cf08233c0cf781a9322e18d75cbd4f41c99c/pkg/base/node_id.go#L82)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

Right, but that's just how it's being used. Functionally it's very similar to `NodeIDContainer` so there's no reason to add extra details here.
</blockquote></details>

Done.
___
*[pkg/storage/pebble.go, line 480 at r3](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-McpP6YPAEpN96zJVPis:-MctL6Gc9-vTMP_pyKPF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/dd55abd3617b6467793674fc51a1bab8f2329fc2/pkg/storage/pebble.go#L480)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

I see what you mean, but it's just requiring the caller to know extra info. All the caller needs to know is "the engine needs a store ID" and the engine's job is "I need to put the store ID in the log's tags". Helps to consolidate logic in one place instead of putting things in multiple places. So yes, I'd advise renaming the method to `SetStoreID`, the interface to `StoreIDSetter`, the init function to `InitStoreIDContainer` and the struct to `StoreIDContainer`. That way if we want to reuse this code for multiple things (eg. sentry tags), it's pretty intuitively clear what each piece is individually doing.
</blockquote></details>

Done.
___
*[pkg/storage/temp_engine.go, line 110 at r4](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-MctCaenC6N1jHAGU42L:-MctL7Uo7lG6aa2tQmWA:b-6qfmsk) ([raw file](https://github.com/cockroachdb/cockroach/blob/30ddf5cb5708f53e911c70cbbae75877a42014a8/pkg/storage/temp_engine.go#L110)):*
<details><summary><i>Previously, itsbilal (Bilal Akhtar) wrote…</i></summary><blockquote>

The usual convention is to type this as `p.(interfaceName).methodName(...)`.  Also this would have to be after the error check below, as if err != nil, p == nil and this would throw a panic.
</blockquote></details>

Done. Got rid of the interface cast here.

**Review by itsbilal — APPROVED** [review]
:lgtm: , you can address the two minor comments below and merge without waiting for a re-review from me. Thanks!

> Reviewed 5 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-:-MctYrZ92yJr3SuSZhli:b-upfc6z)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @bananabrick)

___
*[pkg/base/node_id.go, line 96 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-MctXUHq10SrpHaGT3Gh:-MctXUHq10SrpHaGT3Gi:bliyknl) ([raw file](https://github.com/cockroachdb/cockroach/blob/c11cdb1bb5290f368ebec1c91db5bcabdb22feab/pkg/base/node_id.go#L96)):*
> ```Go
> // which will be initialized later.
> func InitStoreIDContainer() *StoreIDContainer {
> 	s := &StoreIDContainer{}
> ```

Sorry just realized this now, but since the type is exported anyway, the init function is unnecessary and you could just do &base.StoreIDContainer{} elsewhere.
___
*[pkg/base/node_id.go, line 109 at r5](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-MctY_oJAMSmQpfTb27r:-MctY_oJAMSmQpfTb27s:b-crvhop) ([raw file](https://github.com/cockroachdb/cockroach/blob/c11cdb1bb5290f368ebec1c91db5bcabdb22feab/pkg/base/node_id.go#L109)):*
> ```Go
> // SafeFormat implements the redact.SafeFormatter interface.
> func (s *StoreIDContainer) SafeFormat(w redact.SafePrinter, _ rune) {
> 	if s == nil {
> ```

As we discussed on Slack, these are probably unnecessary as (if I read this correctly) s will always be non-nil. Removing it also makes this consistent with NodeIDContainer.

**Review by itsbilal — APPROVED** [review]
:lgtm:

> Reviewed 3 of 3 files at r6.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-:-Mcu8EAG1Ha3Y3fHV0Cr:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @bananabrick)

**Comment by bananabrick** [comment]
TFTR!

bors r+

**Review by jbowens — COMMENTED** [review]
@bananabrick, @itsbilal Can we backport this to 21.1? I'm finding the lack of store ID is a pretty big pain point right now :grimacing: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/66728#-:-MdYLdRZ0KYD0KfM4zIL:bo92lv4)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

**Pull request merged**
